cli/dcoscli/metrics.py

```diff
@@ -0,0 +1,140 @@
+import contextlib
+import json
+
+from dcos import emitting, http, util
+from dcos.errors import DCOSException, DCOSHTTPException
+from dcoscli import tables
+
+logger = util.get_logger(__name__)
+emitter = emitting.FlatEmitter()
+
+
+def _gib(n):
+    return n * pow(2, -30)
+
+
+def _fetch_node_metrics(url):
+    """Retrieve the metrics data from `dcos-metrics`' `node` endpoint.
+
+    :param url: `dcos-metrics` `node` endpoint
+    :type url: str
+    :returns: List of metrics datapoints
+    :rtype: [dict]
+    """
+    with contextlib.closing(http.get(url)) as r:
+
+        if r.status_code == 204:
+            raise DCOSException('No metrics found')
+
+        if r.status_code != 200:
+            raise DCOSHTTPException(r)
+
+        return r.json().get('datapoints', [])
+
+
+def _get_datapoint(datapoints, name, tags=None):
```

Comment:
ah, we need pydoc for private methods? There are going to be quite a few of these. Incoming commit.